### DISCUSSION TITLE
Inject keycloak client secret with vault injector

### DISCUSSION THREAD

**Discussion by thomaswetzler** [argoproj/argo-cd#16345] [Q&A]
Hi,

I try to setup argocd with keycloak authentication. Everything works well when following the Argocd documentation. The referenced keycloak client secret is stored within the argocd-secret. So far, so good. 

I implemented keycloak such a way, that it stores the client secret into a vault cluster. With the vault injector, I‘m able to store the secret within the argocd-server pod in directory /vault/secret. All values are given and correct.
Within argocd-cm I have to configure a reference where to read the client secret like

clientSecret: $argocd-oidc:oidc.azuread.clientSecret

I find only the possibility to read from a k8s secret. 

Regrettably I don‘t find an option how to read it from a file or an environment variable set by secretref in the deployment.

Does anyone know, how to connect that parameter to vault? 

Thanks & BR

**Comment by koroleika** [comment]
We encountered the same problem. It would be much safer to do as the author describes.
